### PR TITLE
Found an unclosed group, closing it

### PR DESCRIPTION
**Pull request by chashiv**
There lies an **unclosed group**, closing it.
Signing off : Shivam Chauhan <Chauhanshivam999@gmail.com>

### CODE REVIEW

**Comment by chashiv** [comment]
> Thank you! Good catch. I think you DCO signature is not correct. Also this is code that is not yet used but a test would be much welcomed, possibly a doctest?

You're Welcome

**Comment by chashiv** [comment]
Updated requested changes with correct DCO and put up a doctest . Also the **is_relative_path** was not returning **bool** values, updated it.

**Comment by pombredanne** [comment]
Thnk you for the updates. Your doctest is failing for now https://travis-ci.org/nexB/scancode-toolkit/jobs/514873900#L1831

**Comment by chashiv** [comment]
> Thnk you for the updates. Your doctest is failing for now https://travis-ci.org/nexB/scancode-toolkit/jobs/514873900#L1831

@pombredanne I will surely do the required changes so as to pass all the cases.

File: src/textcode/strings.py [hunk 1]
```diff
@@ -27,18 +27,16 @@
 
 import re
 import string
+import os
```
Comment (pombredanne) [inline src/textcode/strings.py]: Please sort your imports

File: src/textcode/strings.py [hunk 2]
```diff
@@ -198,9 +196,14 @@ def is_relative_path(s):
...
+    >>> is_relative_path("../usr/lib")
+    True
+    >>> is_relative_path("../usr")
+    True
+    >>> is_relative_path("..")
+    True
     """
-    relative = re.compile('^(?:([^/]|\.\.)[\w_\-]+/.*$', re.IGNORECASE).match
-    return relative(s)
+    return not os.path.isabs(s)
```
Comment (pombredanne) [inline src/textcode/strings.py]: IMHO this cannot work as the paths could be POSIX or Windows and you may be running on either one OSes.

File: src/textcode/strings.py [hunk 3]
```diff
@@ -299,4 +302,4 @@ def filter_strings(strs, nglen=4):
     import sys
     location = sys.argv[1]
     for s in strings_from_file(location):
-        print(s)
+        print(s)
```
Comment (pombredanne) [inline src/textcode/strings.py]: Why did you remove the final LF?

File: src/textcode/strings.py (hunk 2 above)
Comment (pombredanne) [reply src/textcode/strings.py]: os.path is either ntpath or posixpath depending on the OS. So this cannot work here since you do not know:
1. which OS will be running scancode
2. for which OS the path found is
You need a cross OS solution. And `os.path` is not it

Comment (chashiv) [reply src/textcode/strings.py]: Should I detect which os scancode will be running on and implement separate regular expression pattern for each case?

File: src/textcode/strings.py [hunk 4]
```diff
@@ -198,9 +196,14 @@ def is_relative_path(s):
     """
     Return True if s looks like a relative posix path.
     Example: usr/lib/librt.so.1 or ../usr/lib
+    >>> is_relative_path("../usr/lib")
+    True
+    >>> is_relative_path("../usr")
+    True
+    >>> is_relative_path("..")
+    True
```
Comment (singh1114) [inline src/textcode/strings.py]: A test for false will help understand this thing better.

File: src/textcode/strings.py (hunk 2 above)
Comment (pombredanne) [reply src/textcode/strings.py]: Of course not. The whole point is that scancode should be able to detect a path in a binary that came from any OS and that while running on any OS.

**Comment by majurg** [comment]
@chashiv @pombredanne There have been no updates to this for a long time. I will close this shortly if no response from @chashiv

**Comment by pombredanne** [comment]
@chashiv this has gone stale so I am closing this at this time. Please reopen if you want to finish this!
Thank you ++
